### DISCUSSION TITLE
feature request: make background truly transparent

### DISCUSSION THREAD

**Discussion by mansoorbarri** [ghostty-org/ghostty#3581] [Feature Requests, Ideas] [answered]
what i'm looking for: 
![image](https://github.com/user-attachments/assets/33c1791f-e75c-4c9e-b88c-7c8f15cf2754)

but setting `set background-opacity` to lower than 1 seems to set the blur on top of the bg-colour rather than make the background truly transparent. 

what I get with my config: 
![image](https://github.com/user-attachments/assets/a7b88b8e-8b40-4507-b1d5-400f4a21b039)

my config 
```
font-family = "Geist Mono"
window-decoration = false
background-opacity = 0.2
```

**Comment by jmckillican** [comment]
I can't reproduce this on macOS 

<img width="1012" alt="Screenshot 2024-12-27 at 4 46 23 PM" src="https://github.com/user-attachments/assets/7a914fc1-6ac7-4639-ab49-50a5fbb9e871" />

**Comment by mansoorbarri** [comment]
i'm on Debian: 
![image](https://github.com/user-attachments/assets/ff855088-2ed7-4b8b-9f43-93cf223a2fea)

**Comment by SHU-red** [comment]
Same on fedora with hyprland
Background just gehts "lighter"

```
             .',;::::;,'.                 shured@fedora
         .';:cccccccccccc:;,.             -------------
      .;cccccccccccccccccccccc;.          OS: Fedora Linux 41 (Workstation Edition) x86_64
    .:cccccccccccccccccccccccccc:.        Host: MS-7E27 (1.0)
  .;ccccccccccccc;.:dddl:.;ccccccc;.      Kernel: Linux 6.12.6-200.fc41.x86_64
 .:ccccccccccccc;OWMKOOXMWd;ccccccc:.     Uptime: 24 mins
.:ccccccccccccc;KMMc;cc;xMMc;ccccccc:.    Packages: 2742 (rpm), 26 (flatpak), 6 (snap), 46 )
,cccccccccccccc;MMM.;cc;;WW:;cccccccc,    Shell: fish 3.7.0
:cccccccccccccc;MMM.;cccccccccccccccc:    Display (LG ULTRAWIDE): 3440x1440 @ 50 Hz in 34" ]
:ccccccc;oxOOOo;MMM000k.;cccccccccccc:    WM: Hyprland (Wayland)
cccccc;0MMKxdd:;MMMkddc.;cccccccccccc;    Cursor: Adwaita
ccccc;XMO';cccc;MMM.;cccccccccccccccc'    Terminal: kitty 0.38.0
ccccc;MMo;ccccc;MMW.;ccccccccccccccc;     Terminal Font: FiraCode-Regular (10.5pt)
ccccc;0MNc.ccc.xMMd;ccccccccccccccc;      CPU: AMD Ryzen 5 7600X (12) @ 5.45 GHz
cccccc;dNMWXXXWM0:;cccccccccccccc:,       GPU 1: NVIDIA GeForce RTX 3050 6GB [Discrete]
cccccccc;.:odl:.;cccccccccccccc:,.        GPU 2: AMD Raphael [Integrated]
ccccccccccccccccccccccccccccc:'.          Memory: 4.49 GiB / 30.45 GiB (15%)
:ccccccccccccccccccccccc:;,..             Swap: 0 B / 8.00 GiB (0%)
 ':cccccccccccccccc::;,.                  Disk (/): 328.96 GiB / 1.82 TiB (18%) - btrfs
                                          Local IP (enp12s0): 192.168.178.20/24
                                          Locale: en_US.UTF-8
```

**Reply by SHU-red** [reply, marked as answer]
Sorry, just hat to restart ghostty!

Works fine and looks nice!

**Reply by mansoorbarri** [reply]
makes sense, i was hot-reloading. restarting works fine.

**Reply by slovdahl** [reply]
In a future more perfect world ghostty could notify the user that a restart is needed when reloading the config and the `background-opacity` has been changed :slightly_smiling_face: Same for `font-size` (as I mentioned in https://github.com/ghostty-org/ghostty/discussions/4314#discussioncomment-11720141) and probably a few other config options.

**Comment by carere** [comment]
Hello all :)

Can we have blurry background also on Linux ?
I'm on OpenSuse with Gnome 47 (Wayland).
I try to get the same effect I had with Alacritty, where I could set the background transparent & blur.
If anyone has a clue, I'll be happy to test it :)

**Reply by jcollie** [reply]
Unfortunately a blurry background cannot be done from within Ghostty. Some compositors are able to do that for you, but not Gnome.

**Reply by nagarete** [reply]
You can try the Blur my Shell extension from Gnome extension. Go to the application, open Ghostty, add Ghostty to the blurred app, and customize the blur there, also don't forget to reduce the Ghostty opacity first from /.config/ghostty/config first.

**Reply by momoRaza** [reply]
> Hello all :)
> 
> Can we have blurry background also on Linux ? I'm on OpenSuse with Gnome 47 (Wayland). I try to get the same effect I had with Alacritty, where I could set the background transparent & blur. If anyone has a clue, I'll be happy to test it :)

You can use blur my shell extension in gnome and get the blur effect, for some reason taking a screenshot of the desktop ignores the blur

https://github.com/user-attachments/assets/d35ae85d-1f64-4f73-bbde-929cf458902a
